common/include/pcl/PCLHeader.h

```diff
@@ -43,6 +43,12 @@ namespace pcl
     return (out);
   }
 
+  inline bool operator== (const PCLHeader &lhs, const PCLHeader &rhs)
+  {
+    return (&lhs == &rhs) ||
+      (lhs.seq == rhs.seq && lhs.stamp == rhs.stamp && lhs.frame_id == rhs.frame_id);
+  }
+
 } // namespace pcl
```

Comment:
So this is an operator overload for `==` comparison right?
Why don't you use it then?